### PR TITLE
Some_blog: Interrations specs

### PR DESCRIPTION
**Pull request by ambrose-kibet**
## In this milestone we
- [x]  Fix N+1 query problemss
- [x] Add integration spec 

after
b4eager

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
+    @user = User.includes(posts: { comments: :user }).find(params[:user_id])
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb:3]: - [ ] according to the requirements you should add screenshots of the logs before and after the fix to the PR description.

File: spec/features/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    it 'shuould have user1\'s profile picture' do
+      visit users_path
+      expect(page).to have_css("img[src*=\"#{@user1.photo}\"]")
+    end
+
+    it 'Shows the number of a  posts for user 1' do
+      visit users_path
+      expect(page).to have_content(@user1.posts_counter)
+    end
```
Comment (mikethreels) [inline spec/features/users_spec.rb]: - [x] according to the requirements, these tests should check if this data is visible for all users not just for user 1
afbeelding

File: spec/features/users_spec.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    it 'Shows the right content' do
+      visit user_path(@user1)
+      expect(page).to have_content(@post2.title)
+    end
+
+    it 'Should redirect to user posts page' do
+      visit user_path(@user1)
+      click_link('See all posts')
+      expect(page).to have_current_path(user_posts_path(@user1))
```
Comment (mikethreels) [inline spec/features/users_spec.rb]: - [x] for the user show page the below tests are still missing:
afbeelding

File: spec/features/posts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    it 'Shows the right no of comment' do
+      visit user_posts_path(@user1)
+      expect(page).to have_content(@post2.comments_counter)
+    end
+
+    it 'Shows the right no of likes' do
+      visit user_posts_path(@user1)
+      expect(page).to have_content(@post2.likes_counter)
+    end
+
```
Comment (mikethreels) [inline spec/features/posts_spec.rb:50]: - [x] for the post index the below tests are still missing:

afbeelding

File: spec/features/posts_spec.rb [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    it 'Shows the right comments for each user' do
+      visit user_post_path(@post2.user, @post2)
+      expect(page).to have_content(@comment1.text)
+    end
+    it 'Shows the right comments for each user' do
+      visit user_post_path(@post2.user, @post2)
+      expect(page).to have_content(@comment2.text)
+    end
```
Comment (mikethreels) [inline spec/features/posts_spec.rb]: - [x] This is testing the same functionality twice with different data, instead one of these tests should assert if you can see the username of each commentor.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Ambrose, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
- [ ] according to the requirements there should be 1 test file for each view.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (rloterh) [reply app/controllers/posts_controller.rb:3]: This issue is well resolved! ✔️

File: spec/features/users_spec.rb (hunk 2 above)
Comment (rloterh) [reply spec/features/users_spec.rb]: well resolved! ✔️

**Review by Davidosky007 — COMMENTED** [review]
Hi @ambrose-kibet 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* All tests passed 👍 
* No linter errors 👍 
* N+1 problems solved 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**